### ISSUE TITLE
` lxc remote add ....` `error: already exists` 

### ISSUE THREAD

**Issue by kenokabe** [lxc/lxd#623]
I have 2 MS Azure remote servers (Ubuntu14.04 LTS), and 
try to copy a contaner
from      foobar01dev.cloudapp.net:8443
to          foobar01pro.cloudapp.net:8443

So, firstly, I try to add the both remote servers to LXD, but for some reason, I could not add the remote side with an  `error: already exists`, although when I confirm by `lxc remote list`, there is no such a LXD remote name in the list.

```
user@foobar01pro:~$ lxc remote add pro foobar01pro.cloudapp.net:8443
Certificate fingerprint: cb 7b d7 ca 60 ce 89 0a 13 ef xx 42 8c 9f 78 70 a0 17 1d ca 3a 8f 19 e1 9f 55 4f 75 xx xx xx xx
ok (y/n)? y

user@foobar01pro:~$ lxc remote list
local <unix:///var/lib/lxd/unix.socket>
pro <https://foobar01pro.cloudapp.net:8443>

user@foobar01pro:~$ lxc remote add dev foobar01dev.cloudapp.net:8443
Certificate fingerprint: 0b f8 dd 33 a3 e8 bd 5c 5d 3a xx ee 83 21 d2 cf 8a 23 33 0c e5 b4 58 2c c1 6d f5 fe xx xx xx xx
ok (y/n)? y
Admin password for dev: 
error: already exists

user@foobar01pro:~$ lxc remote list
local <unix:///var/lib/lxd/unix.socket>
pro <https://foobar01pro.cloudapp.net:8443>

```

What is wrong with this, and how can I fix the situation?

Thanks.

**Comment by kenokabe** [comment]
OK, just after having post this issue, I know why right now.

Sub-domain, isn't it??
MS Azure server DNS is ***.cloudapp.net
and the current LXD(0.8.1) recognize them all the same.

In fact, I change DNS to IP, and it works. Please fix this issue.

Thanks!

**Comment by benbooth493** [comment]
I've found this a problem too.

Let's say I have a LXD server with hostname images01.somedomain . The primary role of which is to simply serve images to other hosts.
So say we have host01 and host02 both of which will need to be trusted by images01.

On host01 we do:
`$ lxc remote add image-server images01.somedomain`

This goes smoothly and we can now start launch containers built upon the images stored on images01.somedomain with:
`$ lxc launch images:<some_image> container01`

Now we would expect to repeat the same command on host02 but this will fail with: 
`error: already exists`

To investigate this let's run the following on image01.somedomain:
    `$ lxc config trust list`

Oddly we see host01 represented as images01.somedomain
This explains why host02 cannot be added as a trusted client on images01.somedomain as it according to the trust list already exist.

The only way to workaround this problem at the moment is to use the IP Address of images01.somedomain when adding on host02. But then what happens when host03 comes online?

**Comment by tych0** [comment]
Hi,

Thanks for the report.

On Mon, May 11, 2015 at 03:08:47AM -0700, Ben Booth wrote:
> I've found this a problem too.
> 
> Let's say I have a LXD server with hostname images01.somedomain . The primary role of which is to simply serve images to other hosts.
> So say we have host01 and host02 both of which will need to be trusted by images01.
> 
> On host01 we do:
> `$ lxc remote add image-server images01.somedomain`
> 
> This goes smoothly and we can now start launch containers built upon the images stored on images01.somedomain with:
> `$ lxc launch images:<some_image> container01`

I assume you mean image-server here instead of images?

> Now we would expect to repeat the same command on host02 but this will fail with: 
> `error: already exists`
> 
> To investigate this let's run the following on image01.somedomain:
>     `$ lxc config trust list`
> 
> Oddly we see host01 represented as images01.somedomain

Sorry, can you show the exact output from the commands? Thanks.

**Comment by benbooth493** [comment]
I will attempt to give you a detailed report based upon 3 lxd hosts. host01, host02, and images01. host01 and host02 will act as container hosts while images01 will act only as a image server.

```
bbooth@images01:~$ lxc config trust list
bbooth@images01:~$
```

```
bbooth@host01:~$ lxc config trust list`
bbooth@host01:~$
```

```
bbooth@host02:~$ lxc config trust list
bbooth@host02:~$
```

As you can see from the above output, there are currently no trusts set up between any of the 3 hosts.

Now let's add images01 as a remote for host01:

```
bbooth@host01:~$ lxc remote add images 10.3.2.157
Certificate fingerprint: 07 9c 62 b2 c9 c0 82 2b 96 88 73 23 ab 4d fb b2 44 5e b5 71 2d 12 04 34 29 eb df 42 07 b5 63 4f
ok (y/n)? y
Admin password for images:
Client certificate stored at server:  images
bbooth@host01:~$
```

Now let's look at the trust list on image01:
```
bbooth@images01:~$ lxc config trust list
10.3.2.157: ed e2 0e 80 ba ba 1a d5 f5 14 08 cc 0f 0c d7 13 c6 a0 65 92 51 11 94 7f 29 a2 85 0e f2 ac 99 89
bbooth@images01:~$
```

Counter-intuitively the trust list shows the ip address of images01 rather than host01 which it has just trusted.

Now let's add images01 as a remote on host02:
```
bbooth@host02:~$ lxc remote add images 10.3.2.157
Certificate fingerprint: 07 9c 62 b2 c9 c0 82 2b 96 88 73 23 ab 4d fb b2 44 5e b5 71 2d 12 04 34 29 eb df 42 07 b5 63 4f
ok (y/n)? y
Admin password for images:
error: already exists
bbooth@host02:~$
```

Here we find the problem!
We cannot add images01 as the remote address is the same as the trust listed on images01.
It would seem that the trust list on images01 should show the client addresses rather than it's own.

**Comment by tych0** [comment]
Oh, I see. With your example and poking at the code it looks like we're just doing something dumb. I'll try to post a fix today, since I think we're tagging 0.9 today.

**Comment by benbooth493** [comment]
Awesome
